org.osgi.test.assertj/pom.xml

```diff
@@ -80,6 +80,20 @@
 			<plugin>
 				<groupId>biz.aQute.bnd</groupId>
 				<artifactId>bnd-maven-plugin</artifactId>
+				<executions>
+					<!-- Integration Test Configuration -->
+					<execution>
+						<id>bnd-process</id>
+						<goals>
+							<goal>bnd-process</goal>
+						</goals>
+						<configuration>
+							<bnd><![CDATA[
+								Import-Package: org.assertj.core.api.recursive.comparison,*
```

Comment:
Perhaps we should do this in source code instead of here? For example, I put the following in `AbstractBundleAssert` to cause the desired package import.

```java
	static { // SoftAssertions will require this at runtime
		Class<?> importRecursive = RecursiveComparisonConfiguration.class;
	}
```

See also https://github.com/osgi/osgi/issues/100 about making a better way to handle this in the future.

Reply:
Sounds like a plan! Yes, that will make the dependency more obvious.

My gut feeling here actually is that the ultimate solution is to fix it in AssertJ somehow. Perhaps wrap it in a custom classloader that can delegate both to the bundle classloader for the bundle that hosts the original assertion, *and* also delegate to the bundle classloader for assertj-core. That will allow the proxy process to do whatever delegations/transformations it wishes to internally with the classes that are visible to it, without users of the bundle needing to figure out which classes they need to import by trial-and-error.

Reply:
-btw, that link to osgi/osgi#100 goes nowhere? Private repo maybe?

Reply:
> Private repo maybe?

Duh. Yes. It will hopefully go public sooner rather than later. Sorry for the confusion.

The link suggests defining an Import annotation in OSGi that can be used to cause a package import. Basically it will make a "hard" reference to a package without actually having to write code to do it.

Reply:
See https://github.com/bndtools/bnd/pull/4132